### PR TITLE
Fresh audit (from 22 April 2020)

### PR DESCRIPTION
**Pull request by bartsmykla**
/cc @spiffxp @puerco 

Audit reflects changes from: #749 and #756

Signed-off-by: Bart Smykla <bsmykla@vmware.com>

### CODE REVIEW

File: audit/projects/k8s-artifacts-prod-bak/buckets/k8s-artifacts-prod-bak/iam.json [hunk 1]
```diff
@@ -14,6 +14,12 @@
       ],
       "role": "roles/storage.legacyBucketReader"
     },
+    {
+      "members": [
+        "group:k8s-infra-artifact-admins@kubernetes.io"
+      ],
+      "role": "roles/storage.legacyBucketWriter"
+    },
```
Comment (spiffxp) [inline audit/projects/k8s-artifacts-prod-bak/buckets/k8s-artifacts-prod-bak/iam.json:22]: Where did this come from? It seems redundant when this group has legacyBucketOwner permissions

File: audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Bucket Policy Only setting for gs://kubernetes-public_cloudbuild:
```
Comment (spiffxp) [inline audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt:1]: Are we using kubernetes-public as a project for cloudbuild? This seems unexpected to me (this applies to all the files in this dir)

File: audit/projects/k8s-staging-autoscaling/iam.json [hunk 3]
```diff
@@ -40,7 +40,6 @@
     {
       "members": [
         "group:k8s-infra-artifact-admins@kubernetes.io",
-        "group:k8s-infra-release-viewers@kubernetes.io",
```
Comment (spiffxp) [inline audit/projects/k8s-staging-autoscaling/iam.json:43]: Why was this removed?

File: audit/projects/k8s-artifacts-prod-bak/buckets/k8s-artifacts-prod-bak/iam.json (hunk 1 above)
Comment (spiffxp) [reply audit/projects/k8s-artifacts-prod-bak/buckets/k8s-artifacts-prod-bak/iam.json:22]: https://github.com/kubernetes/k8s.io/pull/749 doesn't seem to explicitly set write permission for this user, I'm wondering if this is in-development detritus? Or does assigning the legacyBucketOwner role also assign legacyBucketWriter?

**Comment by spiffxp** [comment]
/assign @thockin 
in case @bartsmykla or @puerco can't answer my questions

File: audit/projects/k8s-staging-autoscaling/iam.json (hunk 3 above)
Comment (bartsmykla) [reply audit/projects/k8s-staging-autoscaling/iam.json:43]: So these was removed by implementing thix fix: https://github.com/kubernetes/k8s.io/pull/756/commits/88c811aa1803cda20e8f1a248515065d80a3a362

File: audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt (hunk 2 above)
Comment (bartsmykla) [reply audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt:1]: That was also confusing for me but I didn't do research about it yet

File: audit/projects/k8s-artifacts-prod/buckets/k8s-artifacts-kind/iam.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    },
+    {
+      "members": [
+        "projectViewer:k8s-artifacts-prod"
+      ],
+      "role": "roles/storage.legacyBucketReader"
+    },
+    {
+      "members": [
+        "group:k8s-infra-artifact-admins@kubernetes.io"
```
Comment (thockin) [inline audit/projects/k8s-artifacts-prod/buckets/k8s-artifacts-kind/iam.json:19]: same bug.  Manually removed and re-ran script to add push-kind

File: audit/projects/k8s-artifacts-prod/buckets/k8s-artifacts-prod/iam.json [hunk 5]
```diff
@@ -14,6 +14,12 @@
       ],
       "role": "roles/storage.legacyBucketReader"
     },
+    {
+      "members": [
+        "group:k8s-infra-artifact-admins@kubernetes.io"
+      ],
+      "role": "roles/storage.legacyBucketWriter"
```
Comment (thockin) [inline audit/projects/k8s-artifacts-prod/buckets/k8s-artifacts-prod/iam.json:21]: Same bug, cleaned up

File: audit/projects/k8s-cip-test-prod/buckets/k8s-cip-test-prod/iam.json [hunk 6]
```diff
@@ -14,6 +14,12 @@
       ],
       "role": "roles/storage.legacyBucketReader"
     },
+    {
+      "members": [
+        "group:k8s-infra-artifact-admins@kubernetes.io"
+      ],
+      "role": "roles/storage.legacyBucketWriter"
```
Comment (thockin) [inline audit/projects/k8s-cip-test-prod/buckets/k8s-cip-test-prod/iam.json:21]: Same bug, cleaned up

File: audit/projects/k8s-gcr-audit-test-prod/buckets/k8s-gcr-audit-test-prod/iam.json [hunk 7]
```diff
@@ -14,6 +14,12 @@
       ],
       "role": "roles/storage.legacyBucketReader"
     },
+    {
+      "members": [
+        "group:k8s-infra-artifact-admins@kubernetes.io"
+      ],
+      "role": "roles/storage.legacyBucketWriter"
```
Comment (thockin) [inline audit/projects/k8s-gcr-audit-test-prod/buckets/k8s-gcr-audit-test-prod/iam.json:21]: Same bug, cleaned up

File: audit/projects/k8s-gcr-backup-test-prod-bak/buckets/k8s-gcr-backup-test-prod-bak/iam.json [hunk 8]
```diff
@@ -14,6 +14,12 @@
       ],
       "role": "roles/storage.legacyBucketReader"
     },
+    {
+      "members": [
+        "group:k8s-infra-artifact-admins@kubernetes.io"
+      ],
+      "role": "roles/storage.legacyBucketWriter"
```
Comment (thockin) [inline audit/projects/k8s-gcr-backup-test-prod-bak/buckets/k8s-gcr-backup-test-prod-bak/iam.json:21]: Same bug, cleaned up

File: audit/projects/k8s-gcr-backup-test-prod/buckets/k8s-gcr-backup-test-prod/iam.json [hunk 9]
```diff
@@ -14,6 +14,12 @@
       ],
       "role": "roles/storage.legacyBucketReader"
     },
+    {
+      "members": [
+        "group:k8s-infra-artifact-admins@kubernetes.io"
+      ],
+      "role": "roles/storage.legacyBucketWriter"
```
Comment (thockin) [inline audit/projects/k8s-gcr-backup-test-prod/buckets/k8s-gcr-backup-test-prod/iam.json:21]: Same bug, cleaned up

File: audit/projects/k8s-release-test-prod/buckets/k8s-release-test-prod/iam.json [hunk 10]
```diff
@@ -19,6 +19,7 @@
     },
     {
       "members": [
+        "group:k8s-infra-artifact-admins@kubernetes.io",
```
Comment (thockin) [inline audit/projects/k8s-release-test-prod/buckets/k8s-release-test-prod/iam.json:22]: Same bug, cleaned up

File: audit/projects/kubernetes-public/iam.json [hunk 11]
```diff
@@ -124,12 +124,6 @@
         "group:k8s-infra-cluster-admins@kubernetes.io"
       ],
       "role": "roles/stackdriver.accounts.viewer"
-    },
-    {
-      "members": [
```
Comment (thockin) [inline audit/projects/kubernetes-public/iam.json:129]: I did this

File: audit/projects/kubernetes-public/services/enabled.txt [hunk 12]
```diff
@@ -4,6 +4,7 @@ bigquery.googleapis.com              BigQuery API
 bigquerystorage.googleapis.com       BigQuery Storage API
 clouderrorreporting.googleapis.com   Error Reporting API
 cloudresourcemanager.googleapis.com  Cloud Resource Manager API
+cloudshell.googleapis.com            Cloud Shell API
```
Comment (thockin) [inline audit/projects/kubernetes-public/services/enabled.txt:7]: <shrug>  I would have thought it was already enabled..

File: audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt (hunk 2 above)
Comment (thockin) [reply audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt:1]: @dims your fingerprints are on this - April 20, 2020 at 5:47:58 PM GMT-7

Can you explain this?  It looks to be something "gcb-docker-gcloud" ?  I suggest we nuke this and be aware of how it came to be

File: audit/projects/k8s-conform/buckets/k8s-conform-capi-openstack/iam.json [hunk 13]
```diff
@@ -15,6 +15,12 @@
       ],
       "role": "roles/storage.legacyBucketReader"
     },
+    {
+      "members": [
+        "group:k8s-infra-conform-capi-openstack@kubernetes.io"
```
Comment (thockin) [inline audit/projects/k8s-conform/buckets/k8s-conform-capi-openstack/iam.json:20]: This seems to be correct, but I don't know why it wasn't there before.  It looks like it was not run when it was added.

@dims - I see in the logs that you manually created service accounts in this project and then generated and downloaded credentials for them (Feb 26).  Can you explain this?

First, we REALLY don't want to be handing out long-lived credentials like that.

Second, we REALLY REALLY do not want to be making manual changes that are not backed by a script.

Third, we REALLY REALLY REALLY don't want people making unilateral changes with no history.  Was there history here?

File: audit/projects/k8s-staging-txtdirect/buckets/k8s-staging-txtdirect-gcb/iam.json [hunk 14]
```diff
@@ -15,6 +15,12 @@
       ],
       "role": "roles/storage.legacyBucketReader"
     },
+    {
```
Comment (thockin) [inline audit/projects/k8s-staging-txtdirect/buckets/k8s-staging-txtdirect-gcb/iam.json:18]: This was, I think, the result of a bug that said reader when it meant writer.

fcd9c914

The readers should be removed.  Doing that now.

File: audit/projects/k8s-conform/buckets/k8s-conform-capi-openstack/iam.json (hunk 13 above)
Comment (dims) [reply audit/projects/k8s-conform/buckets/k8s-conform-capi-openstack/iam.json:20]: @thockin logged an issue here when it was done:
https://github.com/kubernetes/k8s.io/issues/604

more context in:
https://github.com/kubernetes/k8s.io/pull/459

File: audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt (hunk 2 above)
Comment (dims) [reply audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt:1]: Not sure what exactly i did here ... i was working on testing the `gcb-docker-gcloud` image using https://cs.k8s.io/?q=gcb-docker-gcloud&i=nope&files=&repos= and probably forgot to switch from community credentials (davanum@gmail.com) to my office one (dims@vmware.com)

File: audit/projects/k8s-conform/buckets/k8s-conform-capi-openstack/iam.json (hunk 13 above)
Comment (thockin) [reply audit/projects/k8s-conform/buckets/k8s-conform-capi-openstack/iam.json:20]: cool, thanks!  I will ask questions there.

File: audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt (hunk 2 above)
Comment (thockin) [reply audit/projects/kubernetes-public/buckets/kubernetes-public_cloudbuild/bucketpolicyonly.txt:1]: OK. I will nuke it

**Comment by thockin** [comment]
@bartsmykla you can re-run and we can take a fresh look :)

**Comment by spiffxp** [comment]
I opened https://github.com/kubernetes/k8s.io/pull/800 to supersede this because I'd like audit results to land in the repo today and we're out of bart's TZ

**Comment by spiffxp** [comment]
/close 
https://github.com/kubernetes/k8s.io/pull/800 merged
